pandas/core/dtypes/dtypes.py

```diff
@@ -60,24 +63,26 @@ class Registry:
     These are tried in order.
     """
     def __init__(self):
-        self.dtypes = []
+        self.dtypes = []  # type: List[Type[ExtensionDtype]]
 
-    def register(self, dtype):
+    def register(self, dtype: Type['ExtensionDtype']) -> None:
         """
         Parameters
         ----------
         dtype : ExtensionDtype
         """
-        if not issubclass(dtype, (PandasExtensionDtype, ExtensionDtype)):
+        if not issubclass(dtype, ExtensionDtype):
             raise ValueError("can only register pandas extension dtypes")
 
         self.dtypes.append(dtype)
 
-    def find(self, dtype):
+    def find(self,
+             dtype: Union[Type['ExtensionDtype'], str],
```

Comment:
We have a `DType` type defined in pandas._typing - is that suitable here instead?

Reply:
No, ``Dtype`` is either a numpy dtype or a pandas extension Dtype. This registry only has Pandas extension Dtypes registered, so ``Dtype`` would be incorrect.

Reply:
A funny thing regarding _typing.py: Because the types there are not forward-referenced, importing _typing caused a circular import error almost anywhere in the code base. So the types there need to be forward-referenced.

Reply:
Do you know what was causing the circular import? That module is pretty new so not surprising if we hit snags like this but we would want to fix that.

I don't think liberal use of forward types is the best solution. If we need to refactor to make things work would rather do that